headers/xdp/libxdp.h

```diff
@@ -103,4 +103,17 @@ struct xdp_program *xdp_multiprog__hw_prog(const struct xdp_multiprog *mp);
 bool xdp_multiprog__is_legacy(const struct xdp_multiprog *mp);
 int xdp_multiprog__program_count(const struct xdp_multiprog *mp);
 
+struct xdp_program_opts {
+	size_t sz;
+	struct bpf_object *obj;
```

Comment:
What about loading from file? It should also be possible to replace xdp_program__{find,open}_file() with this...

Reply:
For replacing those, the parameters look the same, should we have different triplets of 'filename, prog_name, object_open_opts' for find_file and open_file? I think that wastes too much space.

or maybe, just two more const char * find_filename and open_filename, and a object_open_opts, then reusing prog_name?

Reply:
Yeah, or just a single filename and a 'bool find'?